doc/articles/features/windows-system-power.md

```diff
@@ -6,14 +6,44 @@ uid: Uno.Features.WSPower
 
 ## `PowerManager`
 
+You can use the `Windows.System.Power.PowerManager` class to query the battery and charging status of the device and subscribe to the related events when these change.
+
+**Legend**
+  - ✔️  Supported
+  - 💬 Partially supported (see below for more details)
+  - ❌ Not supported
+
+| Picker         | UWP/WinUI   | WebAssembly | Android | iOS   | macOS | WPF | GTK |
+|----------------|-------|-------------|---------|-------|-------|-----|-----|
+| `BatteryStatus` | ✔️   | ✔️  | ✔️     | ✔️    |❌ ️   | ❌ | ❌ ️  |
+| `EnergySaverStatus` | ✔️   |  ❌ | ✔️     | ✔️    |❌ ️   | ❌ | ❌ ️  |
+| `PowerSupplyStatus` | ✔️   | ✔️  | ✔️     | ✔️   |❌ ️   | ❌ | ❌ ️  |
+| `RemainingChargePercent` | ✔️   | ✔️ | ✔️     | ✔️   |❌ ️   | ❌ | ❌ ️  |
+| `RemainingDischargeTime` | ✔️   | ✔️ |  ❌    | ❌ |❌ ️   | ❌ | ❌ ️  |
+| `BatteryStatusChanged` | ✔️   | ✔️  | ✔️     | ✔️   |❌ ️   | ❌ | ❌ ️  |
+| `EnergySaverStatusChanged` | ✔️   |  ❌ | ✔️     | ✔️    |❌ ️   | ❌ | ❌ ️  |
+| `PowerSupplyStatusChanged` | ✔️   | ✔️  | ✔️     | ✔️   |❌ ️   | ❌ | ❌ ️  |
+| `RemainingChargePercentChanged` | ✔️   | ✔️| ✔️     | ✔️   |❌ ️   | ❌ | ❌ ️  |
+| `RemainingDischargeTimeChanged` | ✔️   | ✔️     |  ❌     |  ❌  |❌ ️   | ❌ | ❌ ️  |
+
+### Usage
+
+For general usage you can follow the documentation provided by [Microsoft](https://learn.microsoft.com/en-us/uwp/api/windows.system.power.powermanager).
+
 ### Limitations
 
-**Android**
-- `RemainingDischargeTime` and `RemainingDischargeTimeChanged` events not supported (platform does not offer appropriate APIs)
-- `RemainingChargePercentChanged` event is not updated continuously as there is no API that provides such events. It is triggered by system `Low` and `Ok` battery state broadcasts only. The `RemainingChargePercent` property always returns the up-to-date value. For continuous monitoring you can set up periodic polling.
+#### Android
+
+`RemainingChargePercentChanged` event is not updated continuously as there is no API that provides such events. It is triggered by system `Low` and `Ok` battery state broadcasts only. The `RemainingChargePercent` property always returns the up-to-date value. For continuous monitoring you can set up periodic polling.
+
+#### WebAssembly
+
+Before any of the properties/events of `PowerManager` can be accessed, the class needs to be initialized via a platform-specific `InitializeAsync` method:
 
-**iOS**
-- `RemainingDischargeTime` and `RemainingDischargeTimeChanged` events not supported (platform does not offer appropriate APIs)
+```csharp
+#if __HAS_UNO__
+var isAvailable = await PowerManager.InitializeAsync();
+#endif
```

Comment:
Why do we need to do this? Is it because the internal APIs are async and WinUI APIs are sync?

Reply:
Yes, that is the case - the class is `static` as a whole and the properties can just be read on Windows automatically. Unfortunately, in case of WASM, the initial retrieval is asynchronous (`navigator.getBattery()` is a promise), so the initialization must happen asynchronously before the properties can be used. For simplicity, I made the `InitializeAsync` method available on all Uno targets and it has a `bool` return that tells you whether the `PowerManager` is available or not.